model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/scaled_dot_product_attention.py

```diff
@@ -0,0 +1,187 @@
+import torch.nn as nn
+import torch
+import math
+from copy import copy
+import numpy as np
+from model_compression_toolkit.core.common.graph.functional_node import FunctionalNode
+from model_compression_toolkit.core.common import BaseSubstitution
+from model_compression_toolkit.core.common.graph.graph_matchers import NodeOperationMatcher
+from model_compression_toolkit.core.common.graph.base_graph import Graph, BaseNode, OutTensor
+from model_compression_toolkit.core.pytorch.constants import DIM
+from model_compression_toolkit.core.pytorch.pytorch_device_config import get_working_device
+
+
+class ScaledDotProductDecomposition(BaseSubstitution):
+    """
+    Removes a scaled_dot_product_attention node from the graph, and replaces it with a compatible graph that
+    consists of:
+    MatMul(over q and transposed k)
+    Mul (for scaling)
+    Add (for masking. optional operation, used in cases that attn_mask ig given)
+    Dropout
+    Softmax
+    Matmul.
+    """
+
+    def __init__(self):
+        """
+        Matches scaled_dot_product_attention node.
+        """
+        super().__init__(matcher_instance=NodeOperationMatcher(nn.functional.scaled_dot_product_attention))
+        self.device = get_working_device()
```

Comment:
I'm not sure we need to deal with device here

Reply:
I'm getting `RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!` when not using device :/
For now I removed `self.device` and using the `get_working_device()` inside `_get_attention_mask_tensor` method.
Do I need to handle the device issue somewhere else instead of inside the substitution?